newrelic-agent/src/main/java/com/newrelic/agent/config/ApplicationLoggingLabelsConfig.java

```diff
@@ -0,0 +1,59 @@
+package com.newrelic.agent.config;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ApplicationLoggingLabelsConfig extends BaseConfig {
+    public static final String ROOT = "labels";
+    public static final String ENABLED = "enabled";
+    public static final String EXCLUDE = "exclude";
+
+    private final boolean enabled;
+    private final Set<String> excludeSet;
+
+    /**
+     * Constructor to initialize LogLabelsConfig from configuration properties.
+     *
+     * @param props Map containing configuration properties.
+     */
+    public ApplicationLoggingLabelsConfig(Map<String, Object> props, String parentRoot) {
+        super(props, parentRoot + ROOT + ".");
+        enabled = props.containsKey(ENABLED) && (boolean) props.get(ENABLED);
+        excludeSet = initExcludes(props);
+    }
+
+    private Set<String> initExcludes(Map<String, Object> props) {
+        List<String> excludeList = (List<String>) props.getOrDefault(EXCLUDE, Collections.emptyList());
+        if (excludeList == null) {
+            return Collections.emptySet();
+        }
+        Set<String> excludeSet = new HashSet<>();
+        for (String exclude : excludeList) {
+            String trimmedExclude = exclude.trim();
+            if (!trimmedExclude.isEmpty()) {
+                excludeSet.add(trimmedExclude);
+            }
+        }
+        return excludeSet;
+    }
+
+    public boolean getEnabled() { return enabled; }
+
+    public Set<String> getExcludeSet() { return excludeSet; }
+
+    public boolean isExcluded(String label) { return getExcludeSet().contains(label); }
+
+    public Map<String, String> removeExcludedLabels(Map<String, String> labels) {
```

Comment:
I might be missing some context, but I think the API could be simplified if the excludes list was applied during the call to `initExcludes` call? (Of course that would assume that you have the labels available to pass into the `ApplicationLoggingLabelsConfig` constructor)

```
ApplicationLoggingLabelsConfig(Map<String, Object> props, String parentRoot, Map<String, String> labels) {
    //stuff
    //call removeExcludedLabels() and make that method privare
}
```